src/main/scala/chisel3/simulator/package.scala

```diff
@@ -117,21 +120,26 @@ package object simulator {
             if (filename.startsWith(workspace.supportArtifactsPath)) {
               filename = filename.substring(workspace.supportArtifactsPath.length + 1)
             }
-            java.nio.file.Files.move(
-              java.nio.file.Paths.get(s"${workspace.supportArtifactsPath}/$filename"),
-              java.nio.file.Paths.get(s"${workspace.primarySourcesPath}/$filename")
-            )
+            val src = java.nio.file.Paths.get(s"${workspace.supportArtifactsPath}/$filename").normalize()
+            if (!movedFiles.contains(src)) {
+              java.nio.file.Files.move(
+                src,
+                java.nio.file.Paths.get(s"${workspace.primarySourcesPath}/$filename")
+              )
+              movedFiles.addOne(src)
+            }
           }
         } finally {
           filelist.close()
         }
       }
 
-      // Move a file in a filelist which may not exist.
+      // Move a file in a filelist which may not exist.  Do nothing if the
+      // filelist does not exist.
       def maybeMoveFiles(filename: String) = try {
         moveFiles(filename)
       } catch {
-        case _ @(_: java.nio.file.NoSuchFileException | _: java.io.FileNotFoundException) =>
+        case _: java.io.FileNotFoundException =>
```

Comment:
Might be a good opportunity for this to check file existence rather than catching exceptions.

Reply:
Switched to that. This was also starting to push things in the direction of this should be using `java.nio.file.Path` instead of strings. I then rewrote the `moveFiles` method to work this way while, I think, preserving the original logic.